app/controllers/internal/badges_controller.rb

```diff
@@ -1,22 +1,29 @@
 class Internal::BadgesController < Internal::ApplicationController
   layout "internal"
+  before_action :ensure_badge, only: :award_badges
 
   def index
     @badges = Badge.all
   end
 
   def award_badges
     usernames = permitted_params[:usernames].split(/\s*,\s*/)
-    badge_slug = permitted_params[:badge]
     message = permitted_params[:message_markdown].presence || "Congrats!"
-    BadgeRewarder.award_badges(usernames, badge_slug, message)
+    BadgeRewarder.award_badges(usernames, permitted_params[:badge], message)
     flash[:success] = "BadgeRewarder task ran!"
     redirect_to internal_badges_url
+  rescue ArgumentError => e
```

Comment:
Hmm I'm testing on development and I'm not hitting the rescue, not sure why though?

Reply:
Huh, that is weird. I followed the same convention for raising as we do in the other internal controllers, and tried changing this to StandardError, but I don't actually get beyond the error to the rescue/flash message. Now I wonder if this this works _anywhere_ in the internal controllers? Perhaps it's not behind handled...?

Reply:
Yeah, I'm at a loss. Putting the `raise` in the action seemed to work for me, but I don't know why a `before_action` would affect this. 🤔

Reply:
@Zhao-Andy Something seems really weird here. I'm pretty sure I've `raise`d from `before_action`s before! In any case, I've just shifted the check into the action since it's the only one we're doing and this is an internal controller.

Thanks for catching this!